### ISSUE TITLE
Sharing iP code quality feedback [for @natlv]

### ISSUE THREAD

**Issue by soc-se-script** [natlv/ip#1]
@natlv We did an automated analysis of your code to detect potential areas to improve the code quality. We are sharing the results below, to help you improve the iP code further.

IMPORTANT: Note that the script looked for **just a few easy-to-detect problems only**, and **at-most three example are given** i.e., there can be other areas/places to improve.

### Aspect: Tab Usage

No easy-to-detect issues :+1: 

### Aspect: Naming boolean variables/methods

No easy-to-detect issues :+1: 

### Aspect: Brace Style

No easy-to-detect issues :+1: 

### Aspect: Package Name Style

No easy-to-detect issues :+1: 

### Aspect: Class Name Style

No easy-to-detect issues :+1: 

### Aspect: Dead Code

No easy-to-detect issues :+1: 

### Aspect: Method Length

No easy-to-detect issues :+1: 

### Aspect: Class size

No easy-to-detect issues :+1: 

### Aspect: Header Comments

Example from [`src/main/java/Natsbot.java` lines `71-75`](https://github.com/natlv/ip/blob/e92f0245016e989f5cd6ab4aa2fc354d1aefe0a0/src/main/java/Natsbot.java#L71..L75):

```java

    /**
     * Main method to launch the Natsbot application.
     *
     * @param args command line arguments
     */

``` 

**Suggestion:** Ensure method/class header comments follow the format specified in the coding standard, in particular, the phrasing of the overview statement.

### Aspect: Recent Git Commit Message

possible problems in commit [`ed99a55`](https://github.com/natlv/ip/commit/ed99a55e42f564bb2e9827195ad047522dee3ec3):

```

Add support for deleting and marking multiple tasks at once

Currently, the delete and mark commands only handle a single task at a time. This limits the efficiency of task management, especially when users need to update multiple tasks simultaneously.

Allowing 'delete' and 'mark' commands to take multiple indices will enhance the user experience by enabling batch operations, reducing the number of commands needed to manage multiple tasks.

Let's:

1. Replace the parseIndex method in CommandParser, which returned a single integer, with parseIndices, which returns a list of integers. This allows parsing multiple indices from the user input.
2. Modify the DeleteCommand and MarkCommand classes to handle multiple indices instead of just one, allowing batch operations for marking and deleting tasks.
3. Update Ui methods showTaskDeleted and showTaskMarked to accept a response string, enabling proper display of messages when multiple tasks are modified.


```

* body not wrapped at 72 characters: e.g., `Currently, the delete and mark commands only handle a single task at a time. This limits the efficiency of task management, especially when users need to update multiple tasks simultaneously.`
            

possible problems in commit [`0d39dee`](https://github.com/natlv/ip/commit/0d39deefe1e66f903c3612719b5f53b8dd4eff7d):

```

Add support for tasks that need to be done after a specific time

Currently, the application does not support tasks that are dependent on specific events or times. This limits the ability to manage tasks that need to be done after a specific time, or after another task or event has occurred.

Adding support for "DoAfter" enables users to specify tasks that are to be completed only after a particular time or event, enhancing task management.

Let's:

1. Create a new DoAfter class to represent tasks that should be done after a specified event or time.
2. Update the CommandParser to parse the new /after syntax and create DoAfterTask instances.
3. Integrate the new task type into the existing command structure to ensure that it is correctly handled and displayed.


```

* body not wrapped at 72 characters: e.g., `Currently, the application does not support tasks that are dependent on specific events or times. This limits the ability to manage tasks that need to be done after a specific time, or after another task or event has occurred.`
            

possible problems in commit [`f8a6ada`](https://github.com/natlv/ip/commit/f8a6ada07c2d7b1dfe2b8ce7eb9ef0633175c072):

```

Set up Continuous Integration (CI) using Github Actions

Currently, the project does not have an automated CI pipeline. This could lead to inconsistent testing and delayed feedback on code quality.

Setting up CI with Github Actions automates the build, test, and validation processes. This ensures code changes are continuously verified and meet quality standards before merging.

Let's:

1. Add a .yml configuration file to define the CI workflow using GitHub Actions.
2. Set up automated triggers on pull requests and commits to run build and test jobs, providing immediate feedback on code changes.


```

* body not wrapped at 72 characters: e.g., `Currently, the project does not have an automated CI pipeline. This could lead to inconsistent testing and delayed feedback on code quality.`
             

**Suggestion:** Follow the given [conventions for Git commit messages](https://se-education.org/guides/conventions/git.html) for _future_ commits (do not modify past commit messages as doing so will change the commit timestamp that we used to detect your commit timings).

### Aspect: Binary files in repo

* [`build\libs\natsbot.jar`](https://github.com/natlv/ip/blob/e92f0245016e989f5cd6ab4aa2fc354d1aefe0a0/build/libs/natsbot.jar) 
             

**Suggestion:** Avoid committing binary files (e.g., `*.class`, `*.jar`, `*.exe`) or third-party library files in to the repo.

----

:information_source: The bot account used to post this issue is un-manned. **Do not reply to this post** (as those replies will not be read). Instead, contact `cs2103@comp.nus.edu.sg` if you want to follow up on this post.
